underfs/s3a/src/main/java/alluxio/underfs/s3a/S3AUnderFileSystem.java

```diff
@@ -588,7 +598,7 @@ private ObjectPermissions getPermissionsInternal() {
 
   @Override
   protected String getRootKey() {
-    return Constants.HEADER_S3A + mBucketName;
```

Comment:
this string concatenation is very __inefficient__ as `getRootKey` gets called on almost every S3 UFS ops, multiple times even. @calvinjia , I would even move this up to `ObjectStoreUFS` to share the optimization across all object stores.